src/mmw/apps/modeling/views.py

```diff
@@ -260,16 +262,20 @@ def _initiate_mapshed_job_chain(mapshed_input, job_id):
     geom = GEOSGeometry(json.dumps(mapshed_input['area_of_interest']),
                         srid=4326)
 
-    chain = (group(geop_tasks(geom, errback, MAGIC_EXCHANGE, choose_worker)) |
-             combine.s().set(exchange=MAGIC_EXCHANGE,
-                             routing_key=choose_worker()) |
-             collect_data.s(geom.geojson).set(link_error=errback,
-                                              exchange=MAGIC_EXCHANGE,
-                                              routing_key=choose_worker()) |
-             save_job_result.s(job_id, mapshed_input)
-             .set(exchange=MAGIC_EXCHANGE, routing_key=choose_worker()))
+    job_chain = (group(
+                 geop_task(t, geom, MAGIC_EXCHANGE, errback, choose_worker)
+                 for t in geop_tasks()) |
+                 combine.s().set(
+                     exchange=MAGIC_EXCHANGE,
+                     routing_key=choose_worker()) |
+                 collect_data.s(geom.geojson).set(
+                     link_error=errback,
+                     exchange=MAGIC_EXCHANGE,
+                     routing_key=choose_worker()) |
+                 save_job_result.s(job_id, mapshed_input).set(
+                     exchange=MAGIC_EXCHANGE, routing_key=choose_worker()))
 
-    return chain.apply_async(link_error=errback)
+    return chain(job_chain).apply_async(link_error=errback)
```

Comment:
I suspect the only reason this was slow originally is because each of those geop tasks calls `choose_worker`. Since you fixed that, we may no longer need to refactor how the job chain is composed.